Clover/app/src/main/java/org/floens/chan/core/manager/WatchManager.java

```diff
@@ -761,6 +761,15 @@ public void onChanLoaderError(ChanLoaderException error) {
         @Override
         public void onChanLoaderData(ChanThread thread) {
             pin.isError = false;
+            /*
+             * Forcibly update watched thread titles
+             * This solves the issue of when you post a thread and the site doesn't have the thread listed yet,
+             * resulting in the thread title being something like /test/918324919 instead of a proper title
+             *
+             * The thread title will be updated as soon as the site has the thread listed in the thread directory
+             * 
+             */
+            pin.loadable.setTitle(PostHelper.getTitle(thread.op, pin.loadable));
```

Comment:
not sure how expensive `.setTitle` is, but maybe we should `.setTitle` only if the current `pin.loadable.title` does not equal `PostHelper.getTitle(thread.op, pin.loadable)`. something like

`if (!pin.loadable.getTitle().equals(PostHelper.getTitle(thread.op, pin.loadable))`

minor thing really I think